src/Tests/dotnet-new.Tests/Utilities.cs

```diff
@@ -24,7 +27,8 @@ internal static string GetTestExecutionTempFolder()
         internal static string CreateTemporaryFolder([CallerMemberName] string caller = "Unnamed", string customName = "")
         {
             string baseDir = Path.Combine(GetTestExecutionTempFolder(), caller, customName);
-            lock (string.Intern(baseDir.ToLowerInvariant()))
+
+            lock (Locker[baseDir.ToLowerInvariant()])
```

Comment:
instead of doing `ToLowerInvariant()` it is possible to use case-insensitive comparison in `ConcurrentDictionary`